internal/handler/handler.go

```diff
@@ -65,31 +88,243 @@ func New(opts *NewOpts) (*Handler, error) {
 		b = oplog.NewBackend(b, opts.L.Named("oplog"))
 	}
 
+	if opts.CappedCleanupPercentage >= 100 || opts.CappedCleanupPercentage <= 0 {
+		return nil, fmt.Errorf(
+			"percentage of documents to cleanup must be in range (0, 100), but %d given",
+			opts.CappedCleanupPercentage,
+		)
+	}
+
 	h := &Handler{
 		b:       b,
 		NewOpts: opts,
 		cursors: cursor.NewRegistry(opts.L.Named("cursors")),
+
+		cappedCleanupStop: make(chan struct{}),
+		cleanupCappedCollectionsDocs: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: namespace,
+				Subsystem: subsystem,
+				Name:      "cleanup_capped_docs",
+				Help:      "Total number of documents deleted in capped collections during cleanup.",
+			},
+			[]string{"db", "collection"},
+		),
+		cleanupCappedCollectionsBytes: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: namespace,
+				Subsystem: subsystem,
+				Name:      "cleanup_capped_bytes",
+				Help:      "Total number of bytes freed in capped collections during cleanup.",
+			},
+			[]string{"db", "collection"},
+		),
 	}
 
 	h.initCommands()
 
+	h.wg.Add(1)
+
+	go func() {
+		defer h.wg.Done()
+
+		h.runCappedCleanup()
+	}()
+
 	return h, nil
 }
 
+// runCapppedCleanup calls capped collections cleanup function according to the given interval.
+func (h *Handler) runCappedCleanup() {
+	if !h.EnableOplog {
+		h.L.Info("The routine to cleanup capped collections is disabled because oplog is disabled")
+		return
+	}
+
+	if h.CappedCleanupInterval <= 0 {
+		h.L.Info("The routine to cleanup capped collections is disabled", zap.Duration("interval", h.CappedCleanupInterval))
+		return
+	}
+
+	ticker := time.NewTicker(h.CappedCleanupInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			if err := h.cleanupAllCappedCollections(context.Background()); err != nil {
```

Comment:
```suggestion
			if err := h.cleanupAllCappedCollections(ctx); err != nil {
```

Otherwise, context.TODO() is, perhaps, more appropriate.